### ISSUE TITLE
Unable to override $zilla->authors

### ISSUE THREAD

**Issue by miyagawa** [rjbs/Dist-Zilla#162]
**tl;dr** - i want to override `$zilla->authors` with the value scanned from POD documents (or git info whatever), and there doesn't seem to be a way.

If `author =` line exists in `dist.ini`, everything is well. We can forget about this situation.

When `author =` doesn't exist, `$zilla->authors` try to pick up from `[%User]` stash in the `~/.dzil/config.ini`, which would give a different value per releaser.

I tried to delete the `[%User]` line from `config.ini`, then dzil tries to get the default author from `copyright_holder`, and to create a copyright holder, it has to pass the default author(!), and it returns an error:

```
[DZ] No %User stash and no copyright holder; can't determine dist author; configure author or a %User section
```

**UPDATE** it might be totally unrelated, since the actual error says it can't find `main_module` - otherwise it could totally find:

```
[DZ] Unable to find main_module in the distribution
[DZ] We tried to guess 'lib/Dist/Zilla/Plugin/LicenseFromModule.pm' but no file like that existed
[DZ] Upon further inspection we didn't find any files in your dist, did you add any?
[DZ] Cannot continue without a main_module
[DZ] No %User stash and no copyright holder; can't determine dist author; configure author or a %User section
```

I hoped that a LicenseProvider plugin could return the holder as the value to override, but `$self->zilla->authors` is a read-only attribute. Also, calling `$self->zilla->authors` inside `provide_license` gets a deep recursion.

Seems like I'm in the dead end, need to override `authors` with a hack, or remove `[%User]` from the stash unless it's new/add action.

Any thoughts?
